src/DefifaGovernor.sol

```diff
@@ -140,12 +157,71 @@ contract DefifaGovernor is
         return super.state(proposalId);
     }
 
+    /**
+     * @dev See {IGovernor-castVoteWithReason}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteWithReason(
+        uint256 proposalId,
+        uint8 support,
+        string calldata reason
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
+    /**
+     * @dev See {IGovernor-castVoteWithReasonAndParams}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteWithReasonAndParams(
+        uint256 proposalId,
+        uint8 support,
+        string calldata reason,
+        bytes memory params
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
+    /**
+     * @dev See {IGovernor-castVoteBySig}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteBySig(
+        uint256 proposalId,
+        uint8 support,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
+
+    /**
+     * @dev See {IGovernor-castVoteWithReasonAndParamsBySig}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteWithReasonAndParamsBySig(
+        uint256 proposalId,
+        uint8 support,
+        string calldata reason,
+        bytes memory params,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public pure override returns (uint256) {
+        revert("use castVote");
```

Comment:
Imo disabling all these methods is not worth it when we can just prevent duplicate tier id's, this would also increase the gas cost for voting substantially as you can no longer specify which tiers should be checked.

Reply:
yeah my thinking was actually based on the test flow that we can use just 1 cast vote method but I get what you are saying